spec.bs

```diff
@@ -1260,8 +1245,35 @@ an [=ordered map=] whose [=map/keys=] are [=strings=] and whose [=map/values=] a
     1. Set |global|'s
       [=InterestGroupBiddingScriptRunnerGlobalScope/is component auction=] to |isComponentAuction|.
     1. Set |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/interest group=] to |ig|.
-    1. Return the result of [=evaluating a script=] with |global|, |script|, "`generateBid`",
+    1. Let |startTime| be the [=current wall time=].
+    1. Let |result| be the result of [=evaluating a script=] with |global|, |script|, "`generateBid`",
       |arguments|, and |timeout|.
+    1. Let |duration| be the [=current wall time=] minus |startTime| in milliseconds.
+    1. If |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/priority=] is not null:
+      1. Set |ig|'s [=interest group/priority=] to |global|'s
+        [=InterestGroupBiddingScriptRunnerGlobalScope/priority=].
+      1. [=list/Replace=] |ig| in the browser’s [=interest group set=].
+    1. If |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/priority signals=]
+      [=map/is not empty=]:
+      1. [=map/For each=] |k| → |v| of |global|'s
+        [=InterestGroupBiddingScriptRunnerGlobalScope/priority signals=]:
+        1. If |v| is null, [=map/remove=] |ig|'s [=interest group/priority signals overrides=][|k|].
+        1. Otherwise, [=map/set=] |ig|'s [=interest group/priority signals overrides=][|k|] to |v|.
+      1. [=list/Replace=] |ig| in the browser’s [=interest group set=].
+    1. If If |result| is not a [=ECMAScript/normal completion=]:
+      1. Set |generatedBid| to |global|'s [=InterestGroupBiddingScriptRunnerGlobalScope/bid=].
+    1. Otherwise:
+      1. Let |generatedBidIDL| be the result of [=converted to an IDL value|converting=]
+        |result|'s \[[Value]] to a {{GenerateBidOutput}}.
+      1. If an exception was caught in the previous step, return failure.
+      1. Let |groupHasAdComponents| be true.
+      1. If |ig|'s [=interest group/ad components=] is null, set |groupHasAdComponents| be false.
+      1. Let |generatedBid| be the result of [=converting GenerateBidOutput to generated bid=]
+        with |generatedBidIDL|, |ig|, |isComponentAuction|, and |groupHasAdComponents|.
+    1. If |generatedBid| is not failure:

```

Comment:
|generatedBid| can also be null via https://pr-preview.s3.amazonaws.com/WICG/turtledove/pull/605.html#dom-interestgroupbiddingscriptrunnerglobalscope-setbid (or maybe as the default, which isn't described in https://pr-preview.s3.amazonaws.com/WICG/turtledove/pull/605.html#interestgroupbiddingscriptrunnerglobalscope-bid).

```suggestion
    1. If |generatedBid| is not failure or null:
```

(But this has downstream problems in https://pr-preview.s3.amazonaws.com/WICG/turtledove/pull/605.html#score-and-rank-a-bid which assumes it's not null.)
or

```suggestion
    1. If |generatedBid| is null, set it to failure.
    1. If |generatedBid| is not failure:
```

or update `setBid()` to ensure it's never null.

Reply:
We should return failure here.